ostd/src/mm/page_table/cursor.rs

```diff
@@ -283,78 +292,47 @@ where
             }
 
             debug_assert!(self.level < self.guard_level);
-            self.level_up();
+            self.pop_level();
         }
     }
 
     pub fn virt_addr(&self) -> Vaddr {
         self.va
     }
 
-    pub fn preempt_guard(&self) -> &DisabledPreemptGuard {
-        &self.preempt_guard
-    }
-
-    /// Goes up a level. We release the current page if it has no mappings since the cursor only moves
-    /// forward. And if needed we will do the final cleanup using this method after re-walk when the
-    /// cursor is dropped.
+    /// Goes up a level.
+    ///
+    /// We release the current page if it has no mappings since the cursor
+    /// only moves forward. And if needed we will do the final cleanup using
+    /// this method after re-walk when the cursor is dropped.
     ///
-    /// This method requires locks acquired before calling it. The discarded level will be unlocked.
-    fn level_up(&mut self) {
+    /// This method requires locks acquired before calling it. The discarded
+    /// level will be unlocked.
+    fn pop_level(&mut self) {
         self.guards[(self.level - 1) as usize] = None;
         self.level += 1;
 
         // TODO: Drop page tables if page tables become empty.
     }
 
-    /// Goes down a level assuming a child page table exists.
-    fn level_down(&mut self) {
+    /// Goes down a level to a child page table.
+    fn push_level(&mut self, child_pt: PageTableNode<E, C>) {
         debug_assert!(self.level > 1);
```

Comment:
`self.level - 1` will be used as an index later, so this debug assertion is meaningless.